ci/tests/general/general_disk.py

```diff
@@ -196,6 +199,32 @@ def partition_disk(disk):
         return disk.partitions[0]
 
     @staticmethod
+    def unpartition_disk(disk, wait=True):
```

Comment:
Always make as generic as possible, meaning i'd provide a partitions parameter too
If partitions is None, partitions = disk.partitions
Do know that disk should become optional too since when you provide partitions, there is (probably) no need for the disk object anymore